### ISSUE TITLE
[Feature Request] Filter things in map editor by difficulty level

### ISSUE THREAD

**Issue by thezerobit** [sirjuddington/SLADE#1618]
**SLADE Version**: (3.2.4)
**Operating System**: (Windows/Linux/OSX)

**Issue Details**:
When adjusting the monsters and powerups for different difficulty levels, sometimes you want a different monster in the same location, or a better powerup, etc. Currently, when there are two things at the same location, it is tricky to work with both of them, you generally have to move one out of the way, to get to the other. If there was an option to only show the things that will appear at a particular difficulty level, it would make this process a lot easier. UDB has a generic filter system with some presets for difficulty levels. I haven't thought of a reason why I would want to do this filtering beyond difficulty flags. I think a way to select between All, Easy, Medium, Hard, (and maybe Multiplayer) things would suffice or cover most use cases.

When a filter is active, only things matching the filter would be displayed on the map, or be selectable.

I don't know exactly what the user interface should be. It could just be small icons that work like radio buttons, or a drop down selector. I am willing to do any/all of the work for this feature, but I'd like some feedback, and maybe input on the design.

**Comment by eevee** [comment]
this is a good thing to have, but i note that "it's hard to work with stacked things" is a more general problem that happens for other reasons too.  improving that seems trickier though

**Comment by sirjuddington** [comment]
Yeah thing (and any other map item in general) filtering is something that's been on the todo list for ages. As for stacked things, there is issue #946 about that, and I'll likely implement something similar to what is suggested there, eventually
